content/platforms/kubernetes/cluster-recovery.md

```diff
@@ -0,0 +1,33 @@
+---
+Title: Cluster Recovery in Kubernetes
+description: 
+weight: 40
+alwaysopen: false
+categories: ["Platforms"]
+aliases: /rs/concepts/kubernetes/cluster-recovery
+---
+
+## Cluster Recovery
+If the cluster losses a majority of its nodes (e.g. from nodes crashing or network split), the cluster will not responde. Recovering the cluster requires performing the following steps:
+
+1. Edit the rec resource, and set the clusterRecovery flag to true, for example by running
+
+> kubectl patch rec <cluster-name> --type merge --patch '{"spec":{"clusterRecovery":true}}'
+
+2. Wait for the cluster to recover, i.e. wait for all the pods to reach a Running status. Tip, to see the progress of the cluster run:
+
+> watch kubectl get pod
+
+3. Recover the data by running the following command from within any cluster pod
+
+> rladmin recover all 
+
+### Notes:
+
+1. For cluster recovery to work, we must be using persistent disks, otherwise there is no way to recvoer the cluster.
+
+2. If you want to restore a cluster from any ccs other than pod-0, you must copy the rdb file to pod-0 and recover from there.
+
```

Comment:
nowhere in our public documentation today do we reference the ccs... we need to provide a little more context or generalize - for example, we can state that recovery in Redis Enterprise is performed from the master node and in Kubernetes we assume that pod <rec-name>-0 is the master node. In rare cases, when the master node is located in another pod and no replication has occurred for a prolonged period of time, it may be necessary to recover from a master node located on a different pod ordinal number. In order to accomplish that, the ccs-redis.rdb file from the /var/opt/redislabs/persist/ccs directory of the master node needs to be copied to the same directory in pod "-0". (what about the other rdb files? what about the difference in file format between AOF and sanpshot? do we need to mention that?)

Reply:
@amiramm , the other ccs-redis.rdb files don't need to be copied, the recovery is performed from a single ccs-redis.rdb file that resides on pod-0.
@danni-m , could the ccs-redis.rdb be an aof format? if yes do we need to mention it?

Reply:
Improved reference to ccs

Reply:
>@danni-m , could the ccs-redis.rdb be an aof format? if yes do we need to mention it?
@otnielvh rdb file can hold both AOF and RDB formats, we don't need to mention it.